src/main/java/org/commoncrawl/spark/util/CCIndex2FilenameParser.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commoncrawl.spark.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CCIndex2FilenameParser {
+    protected static final Pattern filenameAnalyzer = Pattern
+                .compile("^(?:common-crawl/)?crawl-data/([^/]+)/segments/([^/]+)/(crawldiagnostics|robotstxt|warc|wat|wet)/");
+
+                // crawl-data/CC-NEWS/2019/01/CC-NEWS-20190101042830-00057.warc.gz
+    protected static final Pattern newsFilenameAnalyzer = Pattern
+                .compile("^(?:common-crawl/)?crawl-data/CC-NEWS/(\\d+)/(\\d+)/CC-NEWS-(.+)\\.warc\\.gz");
+
+    // Class to encapsulate the extracted crawl, segment, and subset.
+    public static class FilenameParts {
+        public String crawl;
+        public String segment;
+        public String subset;
+    }
+
+    // Error class if we can't find the crawl, segment, and subset.
+    public static class FilenameParseError extends Exception {
+        public FilenameParseError(String message) {
+            super(message);
+        }
+    }
+
+    public static FilenameParts getParts(String filename) throws FilenameParseError {
+        FilenameParts parts = new FilenameParts();
+        Matcher m = filenameAnalyzer.matcher(filename);
+        if(m.find()){
```

Comment:
Nit-picking: this project adds a space before opening parentheses or brackets. The default Eclipse code style is used. Shall we add a code formatter to the build?

Reply:
> default Eclipse code style is used

Ah - sure - do you know (or can you provide public URL of) what the default eclipse formatter settings are?
Is it the google one? or something different?
```https://raw.githubusercontent.com/google/styleguide/gh-pages/eclipse-java-google-style.xml```

Reply:
Or, food for thought/broader discussion - can we use EditorConfig so any IDE can work well in every langauge?
https://editorconfig.org/ Probably not a topic to discuss here though.

Reply:
Thanks for the pointer. I'll have a look. But, yes, let's discuss it separately.